### DISCUSSION TITLE
Can we ASTTransformers like preprocessor for swc/jest Similar to ts-jest

### DISCUSSION THREAD

**Discussion by sunilsurana** [swc-project/swc#7950] [Q&A]
We are moving from ts-jest to swc/jest. We have certain ASTTransformers which are used in TS-jest. Is there anyway to use that with swc/jest

One approach possibly could be create custom jest transformer which first calls asttransformer and then swc/jest. Any other idea?

**Comment by kdy1** [comment]
It will not give you a good performance. I recommend creating Wasm plugins for them.

**Reply by sunilsurana** [reply]
Cool. Thanks for responding. Will the wasm plugin run before swc/jest compiler on each TS file it encounters?

**Reply by kdy1** [reply]
Yeap, but note that it's hard to share a state because transforms may run in parallel.
